File: C/Algorithms/1678.c
#include <string.h>

char *interpret(char *s) {
  int length = strlen(s), i = 0, cur = 0;

  while (i < length) {
    // () case
    if ((*(s + i) == '(') && (*(s + i + 1) == ')')) {
      *(s + cur++) = 'o';
      i = i + 2;
    } else if (*(s + i) == 'G') {
      s[cur++] = 'G';
      i++;
    }
    // case (al)
    else {
      *(s + cur++) = 'a';
      *(s + cur++) = 'l';
      i = i + 4;
    }
  }
  *(s + cur) = '\0';
  return s;
}

File: Rust/Algorithms/1678.rs
impl Solution {
    pub fn interpret(command: String) -> String {
        let length = command.len();
        let mut i = 0;
        let mut result = String::new();
        let mut char_it = command.chars();

        while let Some(c1) = char_it.next() {
            match c1 {
                // Match 'G', nothing else to do.
                'G' => {
                    result.push(c1);
                }
                '(' => {
                    // Cases () and (al)
                    match char_it.next() {
                        // Match '()'
                        Some(')') => {
                            result.push('o');
                        }
                        Some('a') => {
                            result.push_str("al");
                            // Better way to advance?
                            // skip won't do.
                            char_it.next();
                            char_it.next();
                        }
                        _ => {}
                    }
                }
                _ => {}
            }
        }
        result
    }
}
